python/lsst/daf/butler/registry/datasets/byDimensions/_manager.py

```diff
@@ -141,7 +149,7 @@ def register(self, datasetType: DatasetType) -> Tuple[DatasetRecordStorage, bool
             inserted = False
         if inserted and datasetType.isComposite:
             for component in datasetType.storageClass.allComponents():
-                self.register(datasetType.makeComponentDatasetType(component))
+                pass  # self.register(datasetType.makeComponentDatasetType(component))
```

Comment:
Looks like you can now remove this whole block.

Reply:
And should we also add in this function to make sure no one tries to register just a component?  There are probably other places we could put that kind of check, but this is the only one I can think of where it might actually succeed (and mess up the registry) rather than failing with some possibly-less-friendly error.

Reply:
Yes. I did mean to remove that. I can add a check here.